target/linux/ath79/dts/ar9342_ubnt_aircube-ac.dts

```diff
@@ -0,0 +1,120 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+
+#include "ar9344.dtsi"
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+/ {
+	compatible = "ubnt,aircube-ac", "qca,ar9342";
+	model = "Ubiquiti airCube AC";
+
+	keys {
+		compatible = "gpio-keys";
+
+		reset {
+			label = "reset";
+			linux,code = <KEY_RESTART>;
+			gpios = <&gpio 17 GPIO_ACTIVE_LOW>;
+			debounce-interval = <60>;
+		};
+	};
+};
+
+&spi {
+	status = "okay";
+	num-cs = <1>;
+
+	flash@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <25000000>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				label = "u-boot";
+				reg = <0x000000 0x040000>;
+				read-only;
+			};
+
+			partition@40000 {
+				label = "u-boot-env";
+				reg = <0x040000 0x010000>;
+				read-only;
+			};
+
+			partition@50000 {
+				compatible = "denx,uimage";
+				label = "firmware";
+				reg = <0x050000 0xf60000>;
+			};
+
+			partition@fb0000 {
+				label = "cfg";
+				reg = <0xfb0000 0x040000>;
+				read-only;
+			};
+
+			art: partition@ff0000 {
+				label = "art";
+				reg = <0xff0000 0x010000>;
+				read-only;
+			};
+		};
+	};
+};
+
+&ref {
+	clock-frequency = <40000000>;
+};
+
+&uart {
+	status = "okay";
+};
+
+&pcie {
+	status = "okay";
+};
+
+&mdio0 {
+	status = "okay";
+
+	phy-mask = <4>;
+	phy0: ethernet-phy@0 {
+		phy-mode = "rgmii";
+		reg = <0>;
+
+		qca,ar8327-initvals = <
+			0x04 0x07600000 /* PORT0 PAD MODE CTRL */
+			0x7c 0x0000007e /* PORT0_STATUS */
+		>;
+	};
+};
+
+&eth0 {
+	status = "okay";
+
+	/* default for ar934x, except for 1000M and 10M */
+	pll-data = <0x06000000 0x00000101 0x00001313>;
+
+	mtd-mac-address = <&art 0x0>;
+
+	phy-mode = "rgmii";
+	phy-handle = <&phy0>;
+
+	gmac-config {
```

Comment:
@damex did you dump these settings from vendor firmware/U-Boot? At least their U-Boot uses totally different settings, without enabling any of the delays, just setting RGMII mode (bit 0 is set).

Reply:
no, i tried 19.07 branch originally without rgmii delays and could not make it work reliably.
then i pick closest device to it from a tree and see how it is configured (ar9342_ubnt_wa) and try to borrow some.

with that delays it was working just fine for around two weeks on 19.07 and later i migrated to trunk to make a pull request. sorry but i didn't test it without rgmii delays on trunk.

since i don't have a working serial console with me right now, could you please help checking if trunk works without delays?

Reply:
Hi @pepe2k 

I originally made those settings. Only this settings made the aircube ac works.
However, I don't know how to get settings from u-boot, maby I was wrong.

David.

Reply:
I don't know if it's related but I could not get MIB info from `swconfig`:
```
Port 0:
	mib: ???
	enable_eee: ???
	igmp_snooping: 0
	vlan_prio: 0
	pvid: 0
	link: port:0 link:up speed:1000baseT full-duplex txflow rxflow 
```

Every ports show ? behind MIB and enable_eee. AFAIK, that was not the case with Ubiquiti firmware.

Reply:
add this

option ar8xxx_mib_type '0'
option ar8xxx_mib_poll_interval '1000'

to switch configuration in /etc/config/network to enable per port statistics. (enabled mib polling)

https://gist.github.com/damex/48e6d703be1f80072e5c4e034e0e3c5a

Reply:
@ david-barbion @damex these settings have impact on the performance and link quality between main SOC and switch. With the current patch I'm barely able to pass 100 Mbps traffic.

Reply:
@pepe2k 

thank you for suggestion about rgmii delays. i got a spare aircube and serial to try.
i tested without rgmii delays and it works well.

i did wireless test in a loop doing 30s test and result is consistent:
```

iperf -t 30 -c potato.home
------------------------------------------------------------
Client connecting to potato.home, TCP port 5001
TCP window size:  129 KByte (default)
------------------------------------------------------------
[  6] local 192.168.210.50 port 51039 connected with 192.168.110.23 port 5001
[ ID] Interval       Transfer     Bandwidth
[  6]  0.0-30.0 sec  1.07 GBytes   306 Mbits/sec
```

it was around 200Mbit before.


also wired test:

```
iperf -t 30 -c potato.home
------------------------------------------------------------
Client connecting to potato.home, TCP port 5001
TCP window size: 1.05 MByte (default)
------------------------------------------------------------
[  6] local 192.168.110.24 port 51116 connected with 192.168.110.23 port 5001
[ ID] Interval       Transfer     Bandwidth
[  6]  0.0-30.0 sec  3.22 GBytes   923 Mbits/sec

```

it was 700Mbit before.

i just pushed an update.
please check it again.

Reply:
could we please take a look at MR again?

Reply:
@pepe2k could you please have look at the changes again?

Reply:
@damex sorry for the delay, I'm very busy recently but will have a look this week.

Reply:
ping